app/cdap/components/Connections/Browser/ParsingConfigModal/store.tsx

```diff
@@ -126,11 +130,34 @@ export const performInitialLoad = (connection, sampleProperties, entity, dispatc
   const sourcePluginName = sourcePluginNameProperty.description;
 
   let widgetKey;
-  // Get the connection details in order to find the arttifact version
+  // Get the connection details in order to find the artifact version
   // Assume the source plugin is in the same artifact as the connector
   const widgetObs = ConnectionsApi.getConnection(params).pipe(
     switchMap((connectionDetails: IConnectionDetails) => {
       const connectorArtifact = connectionDetails.plugin.artifact;
+      if (connectorArtifact.version) {
+        return Observable.of(connectorArtifact);
+      } else {
+        // If no verion was provided by this response,
+        // look up the artifact and get the latest version
+        // This can happen for the default connection
+        const listArtifactVersionsParams = {
```

Comment:
So why are we deciding this in UI? Theoretically all name, scope and version can be null, and backend should have the logic to pick up the latest one, from the JIRA, I don't get why we want this logic in UI.